vortex-file/src/open.rs

```diff
@@ -1,68 +1,66 @@
-mod exec;
-mod split_by;
-
+use std::marker::PhantomData;
 use std::sync::Arc;
 
-pub use exec::*;
 use flatbuffers::root;
 use futures_util::stream::FuturesUnordered;
 use futures_util::{stream, StreamExt, TryStreamExt};
 use itertools::Itertools;
 use moka::future::CacheBuilder;
-pub use split_by::*;
 use vortex_array::ContextRef;
 use vortex_buffer::{ByteBuffer, ByteBufferMut};
 use vortex_dtype::DType;
 use vortex_error::{vortex_bail, vortex_err, VortexExpect, VortexResult};
 use vortex_flatbuffers::{dtype as fbd, footer as fb, FlatBuffer, ReadFlatBuffer};
 use vortex_io::VortexReadAt;
+use vortex_layout::scan::ScanDriver;
 use vortex_layout::segments::SegmentId;
 use vortex_layout::{Layout, LayoutContextRef, LayoutId};
+use vortex_sampling_compressor::ALL_ENCODINGS_CONTEXT;
 
 use crate::footer::{FileLayout, Postscript, Segment};
-use crate::io::file::FileIoDriver;
 use crate::segments::{InMemorySegmentCache, NoOpSegmentCache, SegmentCache};
-use crate::{VortexFile, EOF_SIZE, MAGIC_BYTES, VERSION};
-
-const INITIAL_READ_SIZE: u64 = 1 << 20; // 1 MB
+use crate::{
+    GenericVortexFile, InMemoryVortexFile, VortexFile, EOF_SIZE, MAGIC_BYTES, MAX_FOOTER_SIZE,
+    VERSION,
+};
+
+pub trait VortexFileOpener: Sized {
```

Comment:
this is a struct in `vortex-datafusion`, how about `VortexFileScanner` or `VortexFileReader`?

Reply:
and as a bigger stylistic question - how about we remove all the `Vortex...` prefix from everything. The users can always alias it.